src/averageAges.js

```diff
@@ -38,6 +48,18 @@ function calculateMenAverageAge(people, century) {
  */
 function calculateWomenAverageAge(people, withChildren) {
   // write code here
+  let women = people.filter(person => person.sex === 'f');
+
+  if (withChildren) {
+    women = women.filter(woman =>
+      people.some(person => person.mother === woman.name)
+    );
+  }
+
+  const ages = women.map(woman => (woman.died - woman.born));
+  const averageAge = ages.reduce((sum, age) => sum + age) / ages.length;
```

Comment:
спробуй рахувати вік в редюсі. Мап в 59 строці зайвий. Це зайва ітерація
```suggestion
  const averageAge = women.reduce((sum, person) => sum + (person.died - person.born), 0) / ages.length;
```